Neos.Flow/Classes/Reflection/ReflectionService.php

```diff
@@ -1224,7 +1224,15 @@ protected function reflectClass($className)
             throw new Exception\InvalidClassException('The class with name "' . $className . '" is a Doctrine proxy. It is not supported to reflect doctrine proxy classes.', 1314944681);
         }
 
-        $class = new ClassReflection($className);
+        // FIXME: This try catch was added with Flow 7.0 to improve the developer experience for upgrading components and can be removed later
+        try {
+            $class = new ClassReflection($className);
+        } catch (\Exception $e) {
+            if (stripos($e->getMessage(), '"Neos\Flow\Http\Component\ComponentInterface"') !== false) {
```

Comment:
I wonder why we need to instantiate the ClassReflection to trigger the exception.. can't we add the check at the place where it was created before?
The check for the exception message is quite quirky, maybe we could instead add the failing class name to the (custom) exception somehow.

Also we should add a `@deprecated` annotation so that we don't forget to remove the hack in a future release

Reply:
> I wonder why we need to instantiate the ClassReflection to trigger the exception.. can't we add the check at the place where it was created before?

Well, the exception happens when the autoloader tries to load the ComponentInterface of course. Since we don't know in advance, which classes might have an `implements ComponentInterface`, we need to somehow include/autoload the current class and catch exceptions. Maybe we could also hook into the autoloading though? That could be a good idea.

Reply:
Okay, I took a look at the seemingly nice `autoloader` approach:
- we need to hardcode the registering of the autoloader somewhere (only for compile time though), so that's not much better than this try catch unless we go the full route to create a new package just for that and do it in the package boot... meh :/
- more importantly: We will not know the class which caused the `ComponentInterface` loading attempt there, so we can not tell that in the error message, which makes it a **lot less useful**

Hence I still think this is a good place to do this check. I adjusted the try catch and the exception so that we don't compare the exception message, but instead can retrieve the class name that caused the reflection loading error from the exception (via static constructor method).